### DISCUSSION TITLE
Question.How do yo fetch paged data from database

### DISCUSSION THREAD

**Discussion by Johnka2** [MarimerLLC/csla#2732] [Questions] [answered]
Have been looking through samples.
How do you implement paged access 
to data using his framework.

**Comment by rockfordlhotka** [comment, marked as answer]
@Johnka2 there used to be a couple different paging samples, but they've faded away I see.

One was pretty cool, for smart clients, as it would do background loading of pages of data while the user was on a page/form.

The other was probably what you are looking for - a server-side web paging model. Fortunately, the answer is pretty straightforward, though there are various ways to do the implementation.

Basically, you want a ReadOnlyListBase subclass that represents the current page of data shown to the user. That part is pretty obvious. The question is how to retrieve pages of data.

```c#
public class MyPage : ReadOnlyListBase<MyPage, MyItem>
{
  [Fetch]
  private void FetchPage(int pageNumber, [Inject] IMyItemDal dal)
  {
    var data = dal.GetPageOfData(pageNumber);
    // load this object with `data`
  }
}
```

You may also want to get the total number of items up front, which can be done, along with getting the first page of data, via a command object:

```c#
public class MyItemListLoader : ReadOnlyBase<MyItemListLoader>
{
  // declare properties for FirstPage (type of MyPage) and PageCount (type of int)

  [Fetch]
  private void Fetch([Inject] IMyItemDal dal, [Inject] IChildDataPortal<MyPage> dp)
  {
    PageCount = dal.GetPageCount;
    FirstPage = dp.Fetch(1); // fetch page 1
  }
}
```

**Comment by Johnka2** [comment]
Your right about what am looking for.
What was the last version the sample existed in and I will try and implement it in current version

**Reply by rockfordlhotka** [reply]
I'm guessing the last version was v4.x - I don't remember seeing it in v5.x.

**Comment by Johnka2** [comment]
I get the idea.

**Comment by Johnka2** [comment]
Thanks
